sunpy/visualization/animator/wcs.py

```diff
@@ -0,0 +1,241 @@
+                    grid = {}
+                coord.grid(**grid)
+
+    def _get_main_axes(self):
+        axes = self.fig.add_axes([0.1, 0.1, 0.8, 0.8], projection=self.wcs,
+                                 slices=self.slices_wcsaxes)
+        return axes
+
+    def plot_start_image(self, ax):
+        if self.plot_dimensionality == 1:
+            artist = self.plot_start_image_1d(ax)
+
+        elif self.plot_dimensionality == 2:
+            artist = self.plot_start_image_2d(ax)
+
+        self._apply_coord_params()
+
+        return artist
+
+    def update_plot(self, val, artist, slider):
+        """
+        Update the plot when a slider changes.
+
+        This method both updates the state of the Animator and also re-draws
+        the matplotlib artist.
+        """
+        ind = int(val)
+        ax_ind = self.slider_axes[slider.slider_ind]
+        self.frame_slice[ax_ind] = ind
+        self.slices_wcsaxes[self.wcs.pixel_n_dim - ax_ind - 1] = ind
+
+        if self.plot_dimensionality == 1:
+            self.update_plot_1d(val, artist, slider)
+        elif self.plot_dimensionality == 2:
+            self.update_plot_2d(val, artist, slider)
+
+        self._apply_coord_params()
+        return super().update_plot(val, artist, slider)
+
+    def plot_start_image_1d(self, ax):
+        """
+        Set up a line plot.
+
+        When plotting with WCSAxes, we always plot against pixel coordinate.
+        """
+        ylim = self.ylim or (self.data.min(), self.data.max())
+        ax.set_ylim(ylim)
+        if self.ylabel:
+            ax.set_ylabel(self.ylabel)
+        line, = ax.plot(self.data[self.frame_index], **self.imshow_kwargs)
+        return line
+
+    @property
+    def data_transposed(self):
+        """
+        Return data for 2D plotting, transposed if needed.
+        """
+        if self.slices_wcsaxes.index('y') < self.slices_wcsaxes.index("x"):
+            return self.data[self.frame_index].transpose()
+        else:
+            return self.data[self.frame_index]
+
+    def update_plot_1d(self, val, line, slider):
+        """
+        Update the line plot.
+        """
+        if val != slider.cval:
+            self.axes.reset_wcs(wcs=self.wcs, slices=self.slices_wcsaxes)
+            line.set_ydata(self.data[self.frame_index])
+            slider.cval = val
+
+    def plot_start_image_2d(self, ax):
+        """
+        Setup an image plot.
+        """
+        imshow_args = {'interpolation': 'nearest',
+                       'origin': 'lower'}
+        imshow_args.update(self.imshow_kwargs)
+        im = ax.imshow(self.data_transposed, **imshow_args)
+        if self.if_colorbar:
+            self._add_colorbar(im)
+        return im
+
+    def update_plot_2d(self, val, im, slider):
+        """
+        Update the image plot.
+        """
+        if val != slider.cval:
+            self.axes.reset_wcs(wcs=self.wcs, slices=self.slices_wcsaxes)
+            im.set_array(self.data_transposed)
```

Comment:
So this means the data is re-transposed every time the plot is updated?  Is this not a performance concern?

Reply:
It does, I don't think transpose is a particularly performance sensitive operation in numpy, as it doesn't modify the data, it just creates a transposed view into the array.